### PR TITLE
Perf fixes and enabling skipped test

### PR DESCRIPTION
**Pull request by DonJayamanne**

### CODE REVIEW

File: src/client/datascience/notebook/helpers/helpers.ts [hunk 1]
```diff
@@ -547,72 +546,105 @@ export function translateCellErrorOutput(output: NotebookCellOutput): nbformat.I
...
     }
 }
 function convertStreamOutput(output: NotebookCellOutput): JupyterOutput {
-    const outputs = output.items
+    const outputs: string[] = [];
+    output.items
         .filter((opit) => opit.mime === CellOutputMimeTypes.stderr || opit.mime === CellOutputMimeTypes.stdout)
         .map((opit) => convertOutputMimeToJupyterOutput(opit.mime, opit.data as Uint8Array) as string)
-        .reduceRight<string[]>((prev, curr) => (Array.isArray(curr) ? prev.concat(...curr) : prev.concat(curr)), []);
+        .forEach((value) => {
```
Comment (DonJayamanne) [inline src/client/datascience/notebook/helpers/helpers.ts:625]: Perf fix

File: src/client/datascience/notebook/helpers/helpers.ts [hunk 2]
```diff
@@ -709,8 +738,7 @@ export function translateCellDisplayOutput(output: NotebookCellOutput): JupyterO
                 unknownOutput.metadata = customMetadata.metadata;
             }
             if (output.items.length > 0) {
-                // eslint-disable-next-line @typescript-eslint/no-explicit-any
-                unknownOutput.data = output.items.reduceRight((prev: any, curr) => {
+                unknownOutput.data = output.items.reduce((prev: any, curr) => {
```
Comment (DonJayamanne) [inline src/client/datascience/notebook/helpers/helpers.ts:741]: reduceRight was wrong, it should be reduce

File: src/test/datascience/notebook/executionService.vscode.test.ts [hunk 3]
```diff
@@ -521,8 +521,7 @@ suite('DataScience - VSCode Notebook - (Execution) (slow)', function () {
             'output from cell 2 should be printed before last background output from cell 1'
         );
     });
-    test('Outputs with support for ansic code `\u001b[A`', async function () {
-        return this.skip();
+    test.only('Outputs with support for ansic code `\u001b[A`', async function () {
```
Comment (DonJayamanne) [inline src/test/datascience/notebook/executionService.vscode.test.ts:524]: This has been fixed upstream in VS COde.

File: src/client/datascience/notebook/helpers/helpers.ts (hunk 1 above)
Comment (DonJayamanne) [reply src/client/datascience/notebook/helpers/helpers.ts:625]: Use of `...` spread operator would result in stack overflow

File: src/client/datascience/notebook/helpers/helpers.ts [hunk 4]
```diff
@@ -547,72 +547,110 @@ export function translateCellErrorOutput(output: NotebookCellOutput): nbformat.I
...
+            const stringValue = textDecoder.decode(value);
             return stringValue;
         }
     } catch (ex) {
         traceError(`Failed to convert ${mime} output from a buffer ${typeof value}, ${value}`, ex);
         return '';
     }
 }
-function convertJupyterOutputToBuffer(mime: string, value: unknown): Buffer {
+function convertJupyterOutputToBuffer(mime: string, value: unknown): NotebookCellOutputItem {
```
Comment (IanMatthewHuff) [inline src/client/datascience/notebook/helpers/helpers.ts:589]: Function could use a name change looks like

**Review by IanMatthewHuff — APPROVED** [review]

**Pull request merged**